libutils/json-utils.c

```diff
@@ -0,0 +1,263 @@
+#include <json-utils.h>
+#include <logging.h>    // Log()
+#include <file_lib.h>   // safe_fopen()
+#include <string_lib.h> // TrimWhitespace()
+#include <csv_parser.h>
+#include <alloc.h>
+#define CF_BUFSIZE 4096
+
+/**
+ * @brief Filters a string, according to env file format
+ *
+ * This is used to parse the part after the equal sign in an env file
+ * Leading and trailing whitespace should already be removed
+ * Will terminate string after closing double quote, even when there's more
+ *
+ * @param src Copy from pointer, can be same as dst
+ * @param dst Copy to pointer, can be same as src
+ * @return beginning of processed string, either dst or dst + 1
+ */
+static char *filtered_copy(const char *src, char *dst)
+{
+    assert(src);
+    assert(dst);
+    char *beginning = dst;
+    char opening_quote = '\0';
+    while (*src != '\0')
+    {
+        if (*src == '\"' || *src == '\'')
+        {
+            if (opening_quote == *src)
+            {
+                break;
+            }
+            else if (opening_quote == '\0')
+            {
+                opening_quote = *src;
+                ++src;
+                if (*src == opening_quote)
+                {
+                    break;
+                }
+            }
+        }
+        if (*src == '\\')
+        {
+            ++src;
```

Comment:
No problem, just add some description and TODO comments in the function docstring so that we know what's missing. Deferring to the tests is not really handy.